includes/class-wc-discounts.php

```diff
@@ -40,6 +40,13 @@ class WC_Discounts {
 	protected $discounts = array();
 
 	/**
+	 * Displayed subtotal of the cart/order
+	 *
+	 * @var integer
+	 */
+	protected $subtotal = 0;
```

Comment:
Rather than store this, let's make a helper method and use the `object` prop.

For carts we can use the get_displayed_subtotal method.
For orders we can use the code you proposed.
If there is no `object`, we can base on price like we do now.

Method could be private and named `get_object_subtotal` or similar and can be used in `validate_coupon_minimum_amount`.